src/arcade/src/Microsoft.DotNet.SignTool/src/ZipDataEntry.cs

```diff
@@ -37,7 +37,9 @@ public ZipDataEntry(string relativePath, Stream contentStream, long contentSize)
                 contentStream.CopyTo(_stream);
             }
 
-            // compute content hash and reset position back
+            // Compute content hash and reset position back
+            // Reset position to 0 prior to content hash computation.
+            _stream.Position = 0;
```

Comment:
Nit: Should this be added only in the else case? I'm not sure if there's ever a scenario where `CanSeek` is true, but the stream doesn't start at position `0` and still has the correct read position. Forcing a reset in that case could potentially cause issues.

Reply:
Could be, true...maybe some odd archive type with a header per file. I'll hoist it.

Reply:
Though, if that were the case, resetting to zero would also cause issues, so it probably doesn't help